src/controller/systemartifact/execution.go

```diff
@@ -0,0 +1,111 @@
+package systemartifact
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/lib/retry"
+	"github.com/goharbor/harbor/src/pkg/systemartifact"
+	"github.com/goharbor/harbor/src/pkg/task"
+	"time"
+)
+
+const (
+	VendorTypeSystemArtifactCleanup = "SYSTEM_ARTIFACT_CLEANUP"
+)
+
+func init() {
+	task.SetExecutionSweeperCount(job.SystemArtifactCleanup, 50)
+}
+
+var Ctl = NewController()
+
+type Controller interface {
+	Start(ctx context.Context, async bool, trigger string) error
+}
+
+func NewController() Controller {
+	return &controller{
+		execMgr:           task.ExecMgr,
+		taskMgr:           task.Mgr,
+		systemArtifactMgr: systemartifact.Mgr,
+		makeCtx:           orm.Context,
+	}
+}
+
+type controller struct {
+	execMgr           task.ExecutionManager
+	taskMgr           task.Manager
+	systemArtifactMgr systemartifact.Manager
+	makeCtx           func() context.Context
+}
+
+func (c *controller) Start(ctx context.Context, async bool, trigger string) error {
+	execId, err := c.execMgr.Create(ctx, VendorTypeSystemArtifactCleanup, 0, trigger)
+	if err != nil {
+		return err
+	}
+	// cleanup job would always be scheduled in async mode in production
+	// allowing for sync mode execution only for test mode purposes
+	// if there are any trigger settings then pass them to the cleanup manager first
+	jobParams := job.Parameters{}
+
+	if !async {
+		err := c.createCleanupTask(ctx, jobParams, execId)
+		if err != nil {
+			return err
+		}
+
+		logger.Info("Created job for scan data export successfully")
+		return nil
+	}
+	go func(ctx context.Context) {
```

Comment:
I'm confusing here why we need a flag `async`, IMO `createCleanupTask` is only create task in db and will not take too long time. And here has a different behavior, in sync mode error will be returned as `Start()` error, but for async mode error only be logged.